server/src/test/java/com/broadcom/lsp/cobol/service/ClientServerIntegrationTest.java

```diff
@@ -0,0 +1,211 @@
+        new CodeActionParams(new TextDocumentIdentifier(UseCaseUtils.DOCUMENT_URI), null, null);
+    CompletionParams completionParams =
+        new CompletionParams(
+            new TextDocumentIdentifier(UseCaseUtils.DOCUMENT_URI), new Position(0, 8));
+
+    Position position = new Position(0, 2);
+    TextDocumentIdentifier testTextDocumentIdentifier =
+        new TextDocumentIdentifier(UseCaseUtils.DOCUMENT_URI);
+    TextDocumentPositionParams positionParams =
+        new TextDocumentPositionParams(testTextDocumentIdentifier, position);
+    ReferenceParams referenceParams = new ReferenceParams();
+
+    assertEquals(SHUTDOWN_RESPONSE, service.codeAction(params).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.definition(positionParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.completion(completionParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.references(referenceParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.documentHighlight(positionParams).get()); // NOSONAR
+    assertEquals(
+        SHUTDOWN_RESPONSE, service.formatting(new DocumentFormattingParams()).get()); // NOSONAR
+
+    List<TextDocumentContentChangeEvent> textEdits = new ArrayList<>();
+    textEdits.add(new TextDocumentContentChangeEvent(INCORRECT_TEXT_EXAMPLE));
+    service.didChange(
+        new DidChangeTextDocumentParams(
+            new VersionedTextDocumentIdentifier(UseCaseUtils.DOCUMENT_URI, 0), textEdits));
+
+    service.didClose(
+        new DidCloseTextDocumentParams(new TextDocumentIdentifier(UseCaseUtils.DOCUMENT_URI)));
+
+    ((TestLanguageServer) server).revokeShutdown();
+    assertEquals(1, ((DisposableLanguageServer) server).getExitCode());
+  }
+
+  @Test
+  void testIncorrectLanguageId() {
+
+    service.didOpen(
+        new DidOpenTextDocumentParams(
+            new TextDocumentItem(UseCaseUtils.DOCUMENT_URI, "incorrectId", 1, TEXT_EXAMPLE)));
+    UseCaseUtils.await(() -> !client.getMessagesToShow().isEmpty());
+    assertTrue(
+        client.getMessagesToShow().stream()
+            .map((Function<MessageParams, Object>) MessageParams::getMessage)
+            .anyMatch(
+                it ->
+                    it.toString()
+                        .equals(
+                            "Cannot find a language engine for the given language ID: incorrectId")));
+  }
+
+  @Test
+  void testNotAllowedFileExtensionAnalysis() {
+    service.didOpen(
+        new DidOpenTextDocumentParams(
+            new TextDocumentItem(CPY_DOCUMENT_URI, LANGUAGE, 1, TEXT_EXAMPLE)));
+    UseCaseUtils.await(() -> !client.getMessagesToLog().isEmpty());
+    assertTrue(
+        client.getMessagesToLog().stream()
+            .map((Function<MessageParams, Object>) MessageParams::getMessage)
+            .anyMatch(
+                it ->
+                    it.toString()
+                        .equals("The given document extension is unsupported: " + CPY_EXTENSION)));
```

Comment:
May be replaced with `Communications` mock to avoid awaiting.

Reply:
This test is covered in CobolTextDocumentServiceTest , where we check that it delegates to communication. And ServerCommunicationsTest has its own unit test cases. Removing these tests as I don't think we need these.